src/plugins/intel_gpu/src/plugin/ops/topk.cpp

```diff
@@ -25,47 +25,11 @@ static void CreateTopKOp(Program& p, const std::shared_ptr<ngraph::op::v1::TopK>
     uint32_t top_k = op->get_k();
     uint64_t chosen_axis = op->get_axis();
 
-    if (op->get_output_size() == 2) {
-        auto mutable_precision = op->get_output_element_type(1);
-        if (mutable_precision == ngraph::element::i64) {
-            mutable_precision = ngraph::element::i32;
+    if (p.use_new_shape_infer() || op->is_dynamic()) {
```

Comment:
if op->is_dyamic(), p.use_new_shape_infer() is always true.

Reply:
updated to check only `use_new_shape_infer`